src/main/resources/include/hail/NativeObj.h

```diff
@@ -39,6 +42,64 @@ class NativeObj :
 // to different off-heap objects - we treat them all as NativeObjPtr.
 using NativeObjPtr = std::shared_ptr<NativeObj>;
 
+// There are at least three different implementations of std::string,
+// one in libc++, one in libstdc++ for abi-version <= 8, and another
+// for abi-version >= 9.  To avoid possible conflicts between prebuilt
+// libraries and dynamically-generated code, we use our own minimal
+// hail::hstring in public interfaces.
+
+class hstring {
```

Comment:
This is going off the rails.  You have to assume the compiler and libraries you pick up when compiling dynamically generated code are compatible with the pre-built objects.  Just used std::string.

Reply:
I think we'll have to discuss this face-to-face this afternoon.

Reply:
As a quick pointer, the code right now does build only one libhail, which happens to be called
"libhail_abi_new" rather than just libhail.

The code has remnants of the test-compiler-version-and-choose-libhail approach, because I was going down that path and then made the minimal changes necessary to get towards a single-libhail which would allow us to get it committed and move on to the more useful PR 
with RowStore changes.

I am reluctant to make substantial design changes - even apparent simplifications - to the
cacheing and module-storage because this is somewhat intricate multi-platform multi-
threaded code.  And the session-to-session cacheing is already about a 12% win on the wall-clock time to run complete tests, just from the 49 distinct NativePackDecoder's generated in the test suite.